### ISSUE TITLE
Bug: consumePurchase() NoSuchMethodError Coroutines

### ISSUE THREAD

**Issue by flobaader** [StephenVinouze/KinApp#1]
Hi,

I wanted to try out your library for my kotlin android app.

But if I want to consume a purchase the following error pops up:

```
FATAL EXCEPTION: main
                                                                                Process: de.selectcode.kinderserienquiz, PID: 10282
                                                                                java.lang.NoSuchMethodError: No static method async$default(Lkotlin/coroutines/experimental/CoroutineContext;ZLkotlin/jvm/functions/Function2;ILjava/lang/Object;)Lkotlinx/coroutines/experimental/Deferred; in class Lkotlinx/coroutines/experimental/DeferredKt; or its super classes (declaration of 'kotlinx.coroutines.experimental.DeferredKt' appears in /data/app/de.selectcode.kinderserienquiz-2/base.apk:classes7.dex)
                                                                                    at com.github.stephenvinouze.core.managers.KinAppManager.consumePurchase(KinAppManager.kt:157)
                                                                                    at de.selectcode.kinderserienquiz.UI.activities.GameView$consumeInAPPPurchase$1.doResume(GameView.kt:165)
                                                                                    at kotlin.coroutines.experimental.jvm.internal.CoroutineImpl.resume(CoroutineImpl.kt:54)
                                                                                    at kotlinx.coroutines.experimental.DispatchedContinuation$resume$1.run(CoroutineDispatcher.kt:171)
                                                                                    at kotlinx.coroutines.experimental.EventLoopImpl$QueuedRunnableTask.invoke(EventLoop.kt:156)
                                                                                    at kotlinx.coroutines.experimental.EventLoopImpl$QueuedRunnableTask.invoke(EventLoop.kt:153)
                                                                                    at kotlinx.coroutines.experimental.EventLoopImpl.processNextEvent(EventLoop.kt:111)
                                                                                    at kotlinx.coroutines.experimental.BlockingCoroutine.joinBlocking(Builders.kt:183)
                                                                                    at kotlinx.coroutines.experimental.BuildersKt.runBlocking(Builders.kt:129)
                                                                                    at kotlinx.coroutines.experimental.BuildersKt.runBlocking$default(Builders.kt:121)
                                                                                    at de.selectcode.kinderserienquiz.UI.activities.GameView.consumeInAPPPurchase(GameView.kt:161)
                                                                                    at de.selectcode.kinderserienquiz.UI.activities.GameView.checkUnconsumedInAPPPurchases(GameView.kt:157)
                                                                                    at de.selectcode.kinderserienquiz.UI.activities.GameView.onBillingReady(GameView.kt:125)
                                                                                    at com.github.stephenvinouze.core.managers.KinAppManager$billingConnection$1.onServiceConnected(KinAppManager.kt:59)
                                                                                    at android.app.LoadedApk$ServiceDispatcher.doConnected(LoadedApk.java:1453)
                                                                                    at android.app.LoadedApk$ServiceDispatcher$RunConnection.run(LoadedApk.java:1481)
                                                                                    at android.os.Handler.handleCallback(Handler.java:751)
                                                                                    at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                                    at android.os.Looper.loop(Looper.java:154)
                                                                                    at android.app.ActivityThread.main(ActivityThread.java:6119)
                                                                                    at java.lang.reflect.Method.invoke(Native Method)
                                                                                    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
                                                                                    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
```

My Dependency versions are:

```
compileSdkVersion 26
buildToolsVersion "26.0.1"
ext.kotlin_version = '1.1.3-2'
```

**Comment by flobaader** [comment]
Also Lint is giving me this warning:

`Warning: com.github.stephenvinouze.core.managers.KinAppManager: can't find referenced method 'kotlinx.coroutines.experimental.Deferred async$default(kotlin.coroutines.experimental.CoroutineContext,boolean,kotlin.jvm.functions.Function2,int,java.lang.Object)' in program class kotlinx.coroutines.experimental.DeferredKt`

**Comment by StephenVinouze** [comment]
Hi @flobaader !
Thanks for the report, I'll look into it. I guess I need to upgrade the coroutine lib.
Also, I'll need to change the way I use coroutines, as I'm doing it wrong. I use the `runBlocking` suspend method which basically blocks the thread and is definitely what we're don't want to happen. I'll probably have to rethink how it behave eventually but in the mean time I can already fix your issue ;)

**Comment by StephenVinouze** [comment]
Hey again @flobaader !
This should be resolved in the `1.0.2` version. I updated the readme in that concern. I also got rid of `runBlocking` to properly handle the async methods

In your case, make sure that in your `build.gradle` file you enable coroutines :

```groovy
kotlin {
    experimental {
        coroutines "enable"
    }
}
```

Also you should trigger your coroutine with `launch(UI)` and call explicitly `wait()` and the suspended methods. The rest of your code will be executed on the UI thread.

Hope it resolved your issue!
